### DISCUSSION TITLE
Compressible Finite Volume HLLC Boundary Conditions

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#18963] [Q&A Modules: Navier-Stokes] [answered]
Does the CNSFVHLLC class have a no-slip boundary condition? In the Incompressible finite volume, I see there is a no-slip wall BC, but I am not sure which of the CNSFVHLLC boundary conditions would mimic that.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think so. @lindsayad 
No slip boundary conditions are more of a viscous flow concept, and the HLLC solver is for Eulerian flows with 0 viscosity.
If you have a reference for their formulation for HLLC we could have a look

Guillaume

**Reply by lindsayad** [reply]
We have this example [here](https://github.com/idaholab/moose/blob/next/modules/navier_stokes/test/tests/finite_volume/cns/natural_convection/average-boussinesq.i) of using our fully compressible Kurganov-Tadmor implementation in a cavity with viscous drag from the walls. Use of the `FVOrthogonalBoundaryDiffusion` objects ensures that the only momentum flux at the walls is due to viscous forces (no advective outflow of momentum).

**Reply by maxnezdyur** [reply]
Thank you. I will look at this right away.

**Reply by GiudGiud** [reply]
Note that a KT boundary condition may not mix well with HLLC kernels. I'd recommend switching over completely from HLLC to KT for this application.

**Reply by lindsayad** [reply]
That example does not use any boundary conditions unique to KT
